go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -454,24 +442,20 @@ func (oc *Controller) deleteEgressFirewallRules(externalID string) error {
 	} else {
 		logicalSwitches = []string{types.OVNJoinSwitch}
 	}
+	sort.Strings(logicalSwitches)
 	stdout, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading", "--columns=_uuid", "find", "ACL",
 		fmt.Sprintf("external-ids:egressFirewall=%s", externalID))
 	if err != nil {
-		return fmt.Errorf("error deleting egressFirewall with external-ids %s, cannot get ACL policies - %s:%s",
+		return nil, fmt.Errorf("error deleting egressFirewall with external-ids %s, cannot get ACL policies - %s:%s",
 			externalID, err, stderr)
 	}
 	uuids := strings.Fields(stdout)
 	for _, logicalSwitch := range logicalSwitches {
 		for _, uuid := range uuids {
-			_, stderr, err := util.RunOVNNbctl("remove", "logical_switch", logicalSwitch, "acls", uuid)
-			if err != nil {
-				deletionErrors = errors.Wrapf(deletionErrors, "failed to delete the ACL rules for "+
-					"egressFirewall with external-ids %s on logical switch %s, stderr: %q (%v)", externalID,
-					types.OVNJoinSwitch, stderr, err)
-			}
+			requests = append(requests, []string{"remove", "logical_switch", logicalSwitch, "acls", uuid})
```

Comment:
@JacobTanenbaum any reason we couldn't just pass the NBTxn into this function and have *it* call AddOrCommit()? Then the caller of `deleteEgressFirewallRules()` would just call Commit().

Reply:
I thought it would be better to return the results then all the potential execution happens at about the same time.

Reply:
done